core/include/detray/builders/homogeneous_volume_material_builder.hpp

```diff
@@ -0,0 +1,82 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "detray/builders/volume_builder.hpp"
+#include "detray/builders/volume_builder_interface.hpp"
+#include "detray/materials/material.hpp"
+#include "detray/materials/predefined_materials.hpp"
+
+// System include(s)
+#include <iostream>
+#include <map>
+#include <memory>
+
+namespace detray {
+
+/// @brief Build a volume with homogeneous volume material.
+///
+/// Decorator class to a volume builder that adds the material data to the
+/// volume while building the volume.
+template <typename detector_t>
+class homogeneous_volume_material_builder final
+    : public volume_decorator<detector_t> {
+
+    public:
+    using scalar_type = typename detector_t::scalar_type;
+
+    /// @param vol_builder volume builder that should be decorated with volume
+    /// material
+    DETRAY_HOST
+    homogeneous_volume_material_builder(
+        std::unique_ptr<volume_builder_interface<detector_t>> vol_builder)
+        : volume_decorator<detector_t>(std::move(vol_builder)) {}
+
+    /// Add all necessary compontents for the volume material
+    ///
+    /// @param mat the material parameters
+    DETRAY_HOST
+    void add_material(material<scalar_type> mat) { m_volume_material = mat; }
```

Comment:
I wonder if `set_material` would be more appropriate looking at the implementation

Reply:
Sure, why not. It will ever only be one material parametrization per volume